### DISCUSSION TITLE
how to restore last focused window after rofi exit ? (i3)

### DISCUSSION THREAD

**Discussion by ghost** [davatorium/rofi#1738] [(Possible) Issues] [answered]
Exiting rofi always removes focus from my existing i3 container & moves completely to another one.
I've tried `steal-focus`, `no-steal-focus` in i3 keybinds beside `-show drun`. But none worked.

What worked, is `-normal-window` which behaves like a dedicated window, & closing it would return focus to the last window I was 
working with.

Lot of applications such as rofimoji, greenclip, also facing the same issue. Exiting their windows also cause focus to go away to a random window. Selecting might work in some cases, not in the most.

(& the github-bot you devs are using in the issue section, is a serious $hit. You might stale, or close the issues, but shouldn't limit all the issues to collaborators. It's not a healthy operation.)

**Comment by DaveDavenport** [comment, marked as answer]
I've used i3 for years, never had the focus change when escaping rofi.
Rofi window should not be managed by i3, and should not change the current focus state (unless you use the window/windowcd mode and explicitly tell rofi to request a change of focus).
Can you pastebin your i3 config and indicate your i3 version?

**Reply by ghost** [reply]
I donno what I did but, i3 doesn't lose focus now. I hope you're correct btw.
I was fiddling around greenclip & rofimoji, then started getting this issue. Right now. Everything's fine. Will let you know if something weird happens again. Tq.
